### PR TITLE
controller check error

### PR DESCRIPTION
**Pull request by subpathdev**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In this line we can see, that the CommTo return nil or an error. https://github.com/kubeedge/kubeedge/blob/master/edge/pkg/devicetwin/dtcontext/dtcontext.go#L71

In this PR I take this error and create an error in function distributeMsg (https://github.com/kubeedge/kubeedge/blob/98c27623689af5d6d70d3f0f4af9c7f6ff3278ba/edge/pkg/devicetwin/dtcontroller.go#L116). This will be catched here: (https://github.com/kubeedge/kubeedge/blob/98c27623689af5d6d70d3f0f4af9c7f6ff3278ba/edge/pkg/devicetwin/dtcontroller.go#L79)

In my opinion we should check if an error occur or not when it is possible. This will also increase the possibility to debug the program.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

### CODE REVIEW

File: edge/pkg/devicetwin/dtcontroller.go [hunk 1]
```diff
@@ -123,7 +123,10 @@ func (dtc *DTController) distributeMsg(m interface{}) error {
 	if message.Msg.GetParentID() != "" {
 		log.LOGGER.Infof("Send msg to the %s module in twin", dtcommon.CommModule)
 		confirmMsg := dttype.DTMessage{Msg: model.NewMessage(message.Msg.GetParentID()), Action: dtcommon.Confirm}
-		dtc.DTContexts.CommTo(dtcommon.CommModule, &confirmMsg)
+		err := dtc.DTContexts.CommTo(dtcommon.CommModule, &confirmMsg)
+		if err != nil {
+			return errors.New("can not communicate with module %v, error message is: %v", dtcommon.CommModule, err)
```
Comment (m1093782566) [inline edge/pkg/devicetwin/dtcontroller.go]: Do not wrap errors, please return the error directly.

File: edge/pkg/devicetwin/dtcontroller.go [hunk 2]
```diff
@@ -135,7 +138,10 @@ func (dtc *DTController) distributeMsg(m interface{}) error {
 	if moduleName, exist := ActionModuleMap[message.Action]; exist {
 		//how to deal write channel error
 		log.LOGGER.Infof("Send msg to the %s module in twin", moduleName)
-		dtc.DTContexts.CommTo(moduleName, &message)
+		err := dtc.DTContexts.CommTo(moduleName, &message)
+		if err != nil {
+			return errors.New("can not communicate with module %v, error message is: %v", moduleName, err)
```
Comment (m1093782566) [inline edge/pkg/devicetwin/dtcontroller.go]: Do not wrap errors, please return the error directly.

File: edge/pkg/devicetwin/dtcontroller.go [hunk 3]
```diff
@@ -123,7 +123,10 @@ func (dtc *DTController) distributeMsg(m interface{}) error {
 	if message.Msg.GetParentID() != "" {
 		log.LOGGER.Infof("Send msg to the %s module in twin", dtcommon.CommModule)
 		confirmMsg := dttype.DTMessage{Msg: model.NewMessage(message.Msg.GetParentID()), Action: dtcommon.Confirm}
-		dtc.DTContexts.CommTo(dtcommon.CommModule, &confirmMsg)
+		err := dtc.DTContexts.CommTo(dtcommon.CommModule, &confirmMsg)
```
Comment (m1093782566) [inline edge/pkg/devicetwin/dtcontroller.go]:
```
if err := xxx {
 xxxx
}
```

To save one line?

File: edge/pkg/devicetwin/dtcontroller.go [hunk 4]
```diff
@@ -135,7 +138,10 @@ func (dtc *DTController) distributeMsg(m interface{}) error {
 	if moduleName, exist := ActionModuleMap[message.Action]; exist {
 		//how to deal write channel error
 		log.LOGGER.Infof("Send msg to the %s module in twin", moduleName)
-		dtc.DTContexts.CommTo(moduleName, &message)
+		err := dtc.DTContexts.CommTo(moduleName, &message)
+		if err != nil {
```
Comment (m1093782566) [inline edge/pkg/devicetwin/dtcontroller.go]:
```
if err := xxx {
 xxxx
}
```
To save one line?

**Comment by m1093782566** [comment]
Squash commits please.

**Comment by m1093782566** [comment]
/lgtm

leave approve to @sids-b @edisonxiang

**Comment by subpathdev** [comment]
I didn't understand why the build failed. (Here is a success travis build on the same changes: https://travis-ci.org/subpathdev/kubeedge )

**Comment by sids-b** [comment]
@subpathdev , you need to make changes in the UT's as this err return would not be there in the code and it will continue to execute

**Comment by sids-b** [comment]
The overall change looks good to me though

File: edge/pkg/devicetwin/dtcontroller_test.go [hunk 5]
```diff
@@ -330,6 +330,24 @@ func TestDTController_distributeMsg(t *testing.T) {
 			}
 		})
 	}
+
+	//Sucess Case
```
Comment (m1093782566) [inline edge/pkg/devicetwin/dtcontroller_test.go]: nit: s/Sucess/Successful

**Comment by sids-b** [comment]
LGTM Overall. Can merge this PR once @m1093782566 review comments fixed

**Comment by m1093782566** [comment]
@subpathdev 

Please squash the commits.

**Comment by m1093782566** [comment]
/approve

THANKS!

**Pull request merged**